### ISSUE TITLE
.NET binding: Socket.Send behavior is unexpected

### ISSUE THREAD

**Issue by CCRcmcpe** [zerotier/libzt#257]
https://github.com/zerotier/libzt/blob/2f0f25a15870ac0ae52815558c539e63d656d506/src/bindings/csharp/Socket.cs#L297-L322

`size` parameter is just... ignored. Sure this would not be surprising.

**Comment by joseph-henry** [comment]
Yes you are correct. I think `size` should replace `(uint)Buffer.ByteLength(buffer)`. If you want to make a PR I'll merge it, otherwise I can do it.

**Comment by chadrockey** [comment]
@joseph-henry This bug still exists, it will send the entire buffer if offered, which really confuses most clients, especially if you have a large storage buffer used....  Can send kB or more.

https://github.com/zerotier/libzt/blob/main/src/bindings/csharp/Socket.cs#L321
